oibackoff/index.d.ts

```diff
@@ -0,0 +1,46 @@
+// Type definitions for oibackoff
+// Project: https://github.com/chilts/oibackoff
+// Definitions by: Joshua DeVinney <https://github.com/geoffreak>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export declare function backoff(opts: BackoffOptions): BackoffInstance;
+
+export declare interface BackoffOptions {
+    maxTries?: number;
+    algorithm?: 'incremental' | 'exponential' | 'fibonacci';
+    delayRatio?: number;
+    maxDelay?: number;
+}
+
+export declare interface BackoffInstance {
+    (fn: Function, callback: Function): void;
```

Comment:
Okay, I'm pretty stuck. Maybe you can find a way to shorten things that actually works? Each gist shows the generated definition of the `BackoffInstance` interface.

[Version 1 script that generates 1800+ lines that work](https://gist.github.com/geoffreak/f175a2c747f7b548c0b1a27e88eef283)
[Version 2 script that generates <300 lines don't work unless you specify the callback generics](https://gist.github.com/geoffreak/82205ea9d6b1be519ea5c813bd72695f)

It seems that as soon as you isolate the generics through an interface, you lose the ability for the compiler to infer the typing.